service/storage_service.cc

```diff
@@ -5951,18 +5951,15 @@ future<service::tablet_operation_repair_result> storage_service::repair_tablet(l
         if (trinfo->stage != locator::tablet_transition_stage::repair) {
             throw std::runtime_error(fmt::format("Tablet {} stage is not at repair", tablet));
         }
-        if (trinfo->session_id) {
-            // TODO: Propagate session id to repair
-            slogger.debug("repair_tablet: tablet={} session_id={}", tablet, trinfo->session_id);
-        } else {
-            throw std::runtime_error(fmt::format("Tablet {} session is not set", tablet));
-        }
+        co_await _group0->group0_server().read_barrier(&_group0_as);
+        topology_guard{topo_guard}.check();
```

Comment:
This seems unnecessary (too eager), it will be checked by the repair handler soon enough.